panther_driver/src/driver_node.py

```diff
@@ -285,51 +325,38 @@ def _roboteq_state_is_correct(self, flags: list) -> bool:
     
         return (self._state_err_no <= 1 / self._driver_state_timer_period)
     
-    def _decode_fault_flag(self, flag_val: int) -> FaultFlag:
-        # For more info see 272-roboteq-controllers-user-manual-v21 p. 246
-        #   https://www.roboteq.com/docman-list/motor-controllers-documents-and-files/
-        ref_flag_val = 0b00000001
-
-        msg = FaultFlag()
-        msg.can_net_err = self._panther_can.can_connection_error()
-        msg.overheat = bool(flag_val & ref_flag_val << 0)
-        msg.overvoltage = bool(flag_val & ref_flag_val << 1)
-        msg.undervoltage = bool(flag_val & ref_flag_val << 2)
-        msg.short_circuit = bool(flag_val & ref_flag_val << 3)
-        msg.emergency_stop = bool(flag_val & ref_flag_val << 4)
-        msg.motor_or_sensor_setup_fault = bool(flag_val & ref_flag_val << 5)
-        msg.mosfet_failure = bool(flag_val & ref_flag_val << 6)
-        msg.default_config_loaded_at_startup = bool(flag_val & ref_flag_val << 7)
-
-        return msg
+    def _decode_flag(
+        self, msg_obj: Union[FaultFlag, RuntimeError, ScriptFlag], msg_fields_list: list, flag_val: int
+    ) -> Union[FaultFlag, RuntimeError, ScriptFlag]:
+        
+        if hasattr(msg_obj, 'can_net_err') and self._panther_can.can_connection_error():
+            msg_obj.can_net_err = True
+            rospy.logerr_throttle(
+                5.0, 
```

Comment:
```suggestion
                10.0, 
```
5 seconds isn't too frequent?